web/src/test/java/io/fabric8/launcher/web/endpoints/osio/OsioEndpointIT.java

```diff
@@ -0,0 +1,229 @@
+                .put("pipelineId", "maven-release")
+                .put("projectName", IMPORT_PROJECT_NAME)
+                .put("spacePath", SPACE)
+                .put("gitOrganization", IMPORT_ORGANIZATION)
+                .put("gitRepository", IMPORT_REPOSITORY)
+                .build();
+        ResponseBodyExtractionOptions validatableResponse = given()
+                .spec(configureOsioEndpoint())
+                .headers(createLaunchHeaders())
+                .formParams(params)
+                .when()
+                .post("/import")
+                .then()
+                .assertThat()
+                .statusCode(200)
+                .body("uuid_link", IsNull.notNullValue())
+                .extract()
+                .body();
+
+        //Then: we receive a status link
+        String uuidLink = validatableResponse.jsonPath()
+                .get("uuid_link").toString();
+
+        //When: we listen for success status
+        CountDownLatch successLatch = getSuccessLatch(uuidLink);
+        successLatch.await(30, TimeUnit.SECONDS);
+        assertThat(successLatch.getCount())
+                //Then
+                .as("The process terminated correctly.")
+                .isZero();
+        cleanUpImport();
+    }
+
+    private Map<String, String> createLaunchHeaders() {
+        return ImmutableMap.<String, String>builder()
+                .put("Authorization", "Bearer " + getOsioIdentity().getToken())
+                .put("X-App", "osio")
+                .put("X-Git-Provider", "GitHub")
+                .build();
+    }
+
+    @Before
+    public void waitUntilEndpointIsReady() {
+        given()
+                .spec(configureCatalogEndpoint())
+                .when()
+                .get("/wait")
+                .then()
+                .assertThat().statusCode(200);
+
+    }
+
+    public void cleanUpLaunch() {
+        final String gitOwner = getGitService().getLoggedUser().getLogin();
+        final String defaultNamespace = getDefaultNamespace();
+        final String launchRepoFullName = GitHelper.createGitRepositoryFullName(gitOwner, LAUNCH_PROJECT_NAME);
+        LOG.info("Going to cleanup launch.");
+        try {
+            getGitService().deleteRepository(launchRepoFullName);
+            LOG.info("Deleted GitHub repository: " + launchRepoFullName);
+            getOpenShiftService().deleteBuildConfig(defaultNamespace, LAUNCH_PROJECT_NAME);
+            LOG.info("Deleted build config in namespace " + defaultNamespace + " named " + LAUNCH_PROJECT_NAME);
+            getOpenShiftService().deleteConfigMap(defaultNamespace, gitOwner);
+            LOG.info("Deleted jenkins config map in namespace " + defaultNamespace + " named " + gitOwner);
+        } catch (final NoSuchRepositoryException nsre) {
+            LOG.severe("Could not remove GitHub repo " + launchRepoFullName + ": " + nsre.getMessage());
+        }
+    }
+
+    public void cleanUpImport() {
```

Comment:
Can we move these cleanup tasks to JUnit Rules?

Reply:
Or maybe as @After and split the import and launch into two tests, since they are separate endpoints anyway